alerts/plugins/ipaddr.py

```diff
@@ -0,0 +1,132 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# Copyright (c) 2014 Mozilla Corporation
+
+from configlib import getConfig, OptionParser
+import netaddr
+import os
+
+
+def isIPv4(ip):
+    try:
+        return netaddr.valid_ipv4(ip)
+    except:
+        return False
+
+
+def isIPv6(ip):
+    try:
+        return netaddr.valid_ipv6(ip)
+    except:
+        return False
+
+
+def addError(message, error):
+    '''add an error note to a message'''
+    if 'errors' not in message.keys():
+        message['errors'] = list()
+    if isinstance(message['errors'], list):
+        message['errors'].append(error)
+
+
+class message(object):
+    def __init__(self):
+        '''
+        takes an incoming alert
+        and uses it to trigger an event using
+        the pager duty event api
+        '''
+
+        # set my own conf file
+        # relative path to the rest index.py file
+        self.configfile = os.path.join(os.path.dirname(__file__), 'ipaddr.conf')
+        self.options = None
+        if os.path.exists(self.configfile):
+            self.initConfiguration()
+
+        self.registration = self.options.keywords.split(" ")
+        self.priority = 1
+
+    def initConfiguration(self):
+        myparser = OptionParser()
+        # setup self.options by sending empty list [] to parse_args
+        (self.options, args) = myparser.parse_args([])
+
+        # fill self.options with plugin-specific options
+        self.options.keywords = getConfig('keywords', 'localhost', self.configfile)
+
+    def onMessage(self, message):
+        '''
+        Examine possible ip addresses for the following:
+          ipv6 in an ipv4 field
+          ipv4 in another field
+          '-' or other invalid ip in the ip field
+        Also sets ipv4 in two fields:
+            ipaddress (decimal mapping IP)
+            ipv4address (string mapping)
+            Elastic search is inconsistent about returning IPs as
+            decimal or IPs.
+            In a query an IP field is returned as string.
+            In a facets an IP field is returned as decimal.
+            No ES field type exists for ipv6, so always having
+            a string version is the most flexible option.
+        '''
+
+        # here is where you do something with the incoming alert message
+        if 'events' in message.keys():
```

Comment:
We're trying to get away from calling `message.keys()`, we can instead do the same logic by `if 'events' in message:`

Reply:
Changed all instances of .keys() to if "a" in b